jhu-idc/idc-isle-dc
# Enable validation for migrations

* Adds `DRUPAL_DEFAULT_MIGRATIONS_VALIDATE` environment variable for switching validation for migrations on or off
  * By default, it is `true`
* Adds `use_env` function for test suites to add environment variables to `.env`, scoped to a single test suite
  * This is used to disable validation for most test suites, via `use_env "DRUPAL_DEFAULT_MIGRATIONS_VALIDATE=false"`
  * The only test suite performing migrations that does _not_ disable validation is `10-migration-backend-tests` 
* Fixes some invalid CSVs in `10-migration-backend-tests`

## To Test, part 1
You should see github tests pass.

Manually, try running a test suite other than `10-migration-backend-tests`.  You should immediately see a message like this fly by:
```
Running 21-large-file-derivatives-nightly using the current Drupal state (i.e. no reset of the environment will occur)

Using additional environment variables and re-loading containers

DRUPAL_DEFAULT_MIGRATIONS_VALIDATE=false
```
`DRUPAL_DEFAULT_MIGRATIONS_VALIDATE` is an environment variable Drupal consults at runtime.  If Drupal is already running in its default state, this will be `true`.  So in order to set it to false, the test mechanics in this PR add `DRUPAL_DEFAULT_MIGRATIONS_VALIDATE=false` to the `.env`, re-load Drupal (so it'll see the `false` value), then restore `.env` to its normal state.  

If you keep watching the screen, notice that you see docker-compose messages fly by, such as:
```
idc_solr_1 is up-to-date
idc_mariadb_1 is up-to-date
Recreating idc_drupal_1 ... done
Waiting for Drupal to start ...
Waiting for Drupal to start ...
Waiting for Drupal to start ...
Waiting for Drupal to start ...
Waiting for Drupal to start ...
Waiting for Drupal to start ...
```

It should leave all functional services untouched except for Drupal.

## To test, part 2

There are nuances related to disabling validation _and_ using the static image.  If you `export CI=true` then run the same test suite you tried in part 1, then you should see an additional message suggesting that the static image is being used.  Once the tests are running, do a `docker ps`, and look at the Drupal image being run.  It should be the static variant!
Resolves jhu-idc/idc-general#368
